// Copyright (c) 2019, The Regents of the University of California
// All rights reserved.
//
// Redistribution and use in source and binary forms, with or without
// modification, are permitted provided that the following conditions are met:
// * Redistributions of source code must retain the above copyright
//   notice, this list of conditions and the following disclaimer.
// * Redistributions in binary form must reproduce the above copyright
//   notice, this list of conditions and the following disclaimer in the
//   documentation and/or other materials provided with the distribution.
// * Neither the name of the University of California nor the
//   names of its contributors may be used to endorse or promote products
//   derived from this software without specific prior written permission.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
// ARE DISCLAIMED. IN NO EVENT SHALL THE REGENTS OF THE UNIVERSITY OF CALIFORNIA
// BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
// POSSIBILITY OF SUCH DAMAGE.

// Next TAG: 4
syntax = "proto3";

package cogrob.service_manager;

enum ResultCode {
  RESULT_UNKNOWN = 0;
  RESULT_OK = 1;
  RESULT_SERVICE_NOT_FOUND = 2;
  RESULT_SERVICE_ALREADY_EXIST = 3;
  RESULT_SERVICE_INVALID_OPTIONS = 4;
  RESULT_SERVICE_TYPE_NOT_SUPPORTED = 5;
  RESULT_NOT_IMPLEMENTED = 6;
  RESULT_INTERNAL = 7;
  RESULT_UPDATE_NO_CHANGE = 8;
  RESULT_SERVICE_UNSUPPORTED_OPTIONS = 9;
  RESULT_SERVICE_REQUEST_NOT_EXIST = 10;
  RESULT_INVALID_SERVICE_ID = 11;
  RESULT_SERVICE_NOT_ACTIVE = 12;
}
